src/main/scala/com/gu/contentapi/firehose/ContentApiFirehoseConsumer.scala

```diff
@@ -0,0 +1,51 @@
+package com.gu.contentapi.firehose
+
+import com.amazonaws.auth.AWSCredentialsProvider
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.{ IRecordProcessor, IRecordProcessorFactory }
+import com.gu.contentapi.firehose.client.PublicationLogic
+import com.gu.contentapi.firehose.kinesis.{ EventProcessor, KinesisStreamReader, SingleEventProcessor }
+import com.gu.crier.model.event.v1.EventPayload.UnknownUnionField
+import com.gu.crier.model.event.v1.EventType.EnumUnknownEventType
+import com.gu.crier.model.event.v1.{ Event, EventPayload, EventType }
+
+class ContentApiFirehoseConsumer(
+    val app: String,
+    val mode: String,
+    val suffix: Option[String],
+    val streamName: String,
+    val stage: String,
+    val kinesisCredentialsProvider: AWSCredentialsProvider,
+    val dynamoCredentialsProvider: AWSCredentialsProvider,
+    val awsRegion: String,
+    val logic: PublicationLogic
+) extends KinesisStreamReader {
+
+  val eventProcessorFactory = new IRecordProcessorFactory {
+    override def createProcessor(): IRecordProcessor = new CapiEventProcessor(logic)
+  }
+}
+
+class CapiEventProcessor(publicationLogic: PublicationLogic) extends EventProcessor[Event] with SingleEventProcessor[Event] {
+
+  val codec = Event
+
+  override protected def processEvent(eventWithSize: EventProcessor.EventWithSize[Event]): Unit = {
+    val event = eventWithSize.event
```

Comment:
We could simplify things by getting rid of the `EventWithSize` class and just passing in `Event` here.